tests/test_dependencies.py

```diff
@@ -101,21 +98,150 @@ def test_job_dependency(self):
         # Test dependant is enqueued at front
         q.empty()
         parent_job = q.enqueue(say_hello)
-        q.enqueue(
-            say_hello,
-            job_id='fake_job_id_1',
-            depends_on=Dependency(jobs=[parent_job])
-        )
-        q.enqueue(
-            say_hello,
-            job_id='fake_job_id_2',
-            depends_on=Dependency(jobs=[parent_job],enqueue_at_front=True)
-        )
-        #q.enqueue(say_hello) # This is a filler job that will act as a separator for jobs, one will be enqueued at front while the other one at the end of the queue
+        q.enqueue(say_hello, job_id='fake_job_id_1', depends_on=Dependency(jobs=[parent_job]))
+        q.enqueue(say_hello, job_id='fake_job_id_2', depends_on=Dependency(jobs=[parent_job], enqueue_at_front=True))
+        # q.enqueue(say_hello) # This is a filler job that will act as a separator for jobs, one will be enqueued at front while the other one at the end of the queue
         w.work(burst=True, max_jobs=1)
 
         self.assertEqual(q.job_ids, ["fake_job_id_2", "fake_job_id_1"])
 
+    def test_multiple_jobs_with_dependencies(self):
+        """Enqueue dependent jobs only when appropriate"""
+        q = Queue(connection=self.testconn)
+        w = SimpleWorker([q], connection=q.connection)
+
+        # enqueue dependent job when parent successfully finishes
+        parent_job = q.enqueue(say_hello)
+        job1 = Queue.prepare_data(say_hello, depends_on=parent_job)
+        job2 = Queue.prepare_data(say_hello, depends_on=parent_job)
+
+        jobs = q.enqueue_many([job1, job2])
+        w.work(burst=True)
+        job1 = Job.fetch(jobs[0].id, connection=self.testconn)
+        job2 = Job.fetch(jobs[1].id, connection=self.testconn)
+        self.assertEqual(job1.get_status(), JobStatus.FINISHED)
+        self.assertEqual(job2.get_status(), JobStatus.FINISHED)
+        q.empty()
+
+        # don't enqueue dependent job when parent fails
+        parent_job = q.enqueue(div_by_zero)
+        job1 = Queue.prepare_data(say_hello, depends_on=parent_job)
+        job2 = Queue.prepare_data(say_hello, depends_on=parent_job)
+
+        jobs = q.enqueue_many([job1, job2])
+        w.work(burst=True)
```

Comment:
I think we don't need to retest whether jobs get enqueued if parent job succeeds or fails, this is already tested else where. For this PR, we simply need to test that jobs (with and without dependencies) are properly created when created with `enqueue_many`.

Reply:
I removed the redundant tests, just left the one that checks the job statuses